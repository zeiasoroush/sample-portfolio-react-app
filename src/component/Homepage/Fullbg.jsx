import React, { Component } from 'react';

class Fullbg extends Component {
  constructor(props) {
    super(props);
    this.state = { count: 1 };

    this.isActiveImg = this.isActiveImg.bind(this);
  }

  isActiveImg(index, ActiveimgID) {
    var id = this.props.activeimgId;
    if (id === '0' && index === 0) {
      return 'active-img';
    } else {
      return ActiveimgID === id ? 'active-img' : '';
    }
  }
  render() {
    return (
      <ul>
        {this.props.contentList.map((item, index) => {
          return (
            <li className={this.isActiveImg(index, item.num)} key={index}>
              <div className="slide-caption">
                <h6 style={{ color: item.color }}>{item.title}</h6>
                <h2 style={{ color: item.color }}>
                  {item.first}
                  <br /> {item.second}
                </h2>
              </div>
            </li>
          );
        })
        }
      </ul >
    );
  }
}

export default Fullbg;

