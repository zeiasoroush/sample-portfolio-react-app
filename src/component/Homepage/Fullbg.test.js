import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Fullbg from './Fullbg';
describe('it should render homepage thumbnail', () => {
  let component, props, isActiveImg;

  beforeEach(() => {
    props = {
      contentItem: [
        {
          num: 1,
          active: true,
          title: 'New Arrival',
          first: 'Beautiful',
          second: 'Earth'
        },
        {
          num: 2,
          title: 'Latest One',
          first: 'Tech',
          second: 'Meeting'
        },
        {
          num: 3,
          title: 'Your Update',
          first: 'Smart',
          second: 'Devices'
        }
      ]
    };

    isActiveImg = jest.fn();

    component = shallow(
      <Fullbg contentList={props.contentItem} className={isActiveImg} />
    );
  });
  it('should render fullbg', () => {
    expect(component).toMatchSnapshot();
  });

  it('should set active background', () => {
    let wrapper = mount(
      <Fullbg contentList={props.contentItem} className={isActiveImg} />
    );
    for (let index = 0; index < 3; index++) {
      const s = wrapper
        .find('li')
        .at(index)
        .hasClass('active');

      if (s) {
        expect(s).toEqual(true);
      } else {
        expect(s).toEqual(false);
      }
    }
  });
});
