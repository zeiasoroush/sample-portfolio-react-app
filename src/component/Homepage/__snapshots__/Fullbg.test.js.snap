// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render homepage thumbnail should render fullbg 1`] = `ShallowWrapper {}`;
